Cython/Compiler/ExprNodes.py

```diff
@@ -2710,10 +2713,13 @@ class IteratorNode(ExprNode):
     counter_cname = None
     reversed = False      # currently only used for list/tuple types (see Optimize.py)
     is_async = False
+    outer_scope = None    # non-None if sequence should be evaluated in a different scope
```

Comment:
This makes it feel a bit like `ScopedExprNode`. The latter does more if you set `.has_local_scope`, but without that, if you set an explicit scope from outside, it might do what you want here.